goal_src/jak2/engine/target/logic-target.gc

```diff
@@ -2737,7 +2734,9 @@
             )
           0
           )
-        (set! (-> self control reaction) target-collision-reaction)
+        (set! (-> self control reaction)
+              (the-as (function control-info collide-query vector cshape-moving-flags) target-collision-reaction)
```

Comment:
same here about function type. I think adding another `vector` argument to end will get rid of this cast.